src/frequenz/sdk/timeseries/logical_meter/_resampled_formula_builder.py

```diff
@@ -40,21 +40,22 @@ def __init__(
         self._metric_id = metric_id
         super().__init__()
 
-    async def _get_resampled_receiver(self, component_id: int) -> Receiver[Sample]:
+    async def _get_resampled_receiver(
+        self, component_id: int, metric_id: ComponentMetricId
+    ) -> Receiver[Sample]:
         """Get a receiver with the resampled data for the given component id.
 
         This receiver would contain data for the `metric_id` specified when creating the
         `ResampledFormulaBuilder` instance.
 
         Args:
             component_id: The component id for which to get a resampled data receiver.
+            metric_id: A metric ID to fetch for all components in this formula.
 
         Returns:
             A receiver to stream resampled data for the given component id.
         """
-        request = ComponentMetricRequest(
-            self._namespace, component_id, self._metric_id, None
-        )
+        request = ComponentMetricRequest(self._namespace, component_id, metric_id, None)
```

Comment:
self._metric_id is not used anymore. Should probably be removed

Reply:
no,  there's one place it is getting used,  in the `push_component_metric` method below.